### PR TITLE
Mosin Rework 

### PR DESCRIPTION
**Pull request by VividPups**
## About the PR
Rework the Mosin to allow Melee now, Butcher and Throwing 

## Why / Balance
Mosin is regarded as one of the worst weapons in the game to the point nearly no one uses it. Countless times people had asked/requested to remove the bayonet from the Mosin sprite. or give it the ability to use the weapon  as an melee weapon and because of how Upstream changed/removed the coding for melee/range weapons it is a bit funky but I feel like it isn't that bad to the point where I would say it's terrible. 
1) Lower the Ammo capacity by 10 -> 5. Five is the IRL Standard ammo capacity for the mosin
2) Raised the Firing rate to 1  
3) Melee and Throw damage is the same.
    Attack rate of .85
         -  Piercing 10 (you are stabbing someone so it 
         - Slash 5  (you are pulling the bayonet out so it causes bleeding

         
## How to test
Spawn in the Mosin.  Shooting the rifle will now be Right click, where a  "pinpoint" melee attack is left click   however when the Ammo for the mosin runs dry the right click turns into a thrust attack  

- [X] I have added screenshots/videos to this PR showcasing its changes in-game, **or** this PR does not require an in-game showcase

## Breaking changes
N/A

**Changelog**

Turns out the Bayonet had a sheath on it 
:cl:
- tweak: Turns out Mosin's Bayonet had a Sheath on it

### CODE REVIEW

**Comment by VividPups** [comment]
Well be working on more of it soon

**Comment by VividPups** [comment]
The only thing Im not like about this is now since the YML fixs. it is showing the left attack with an Animation when. the attack itself relies on where you click. I however, do not know how to fix that

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 1]
```diff
@@ -38,16 +38,60 @@
   - type: StaticPrice
     price: 500
 
-- type: entity
+- type: entity #Frontier
   name: Kardashev-Mosin
   parent: BaseWeaponSniper
   id: WeaponSniperMosin
-  description: A civilian grade weapon for hunting, or endless trench warfare. Uses .30 rifle ammo.
+  description: A civilian grade weapon for hunting, or endless trench warfare. Uses .30 rifle ammo. Now with new shiney bayonet!
```
Comment (whatston3) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:45]: Reads a bit strangely, as if this bayonet is a new device in-universe.  As an alternative, I'd put this forth:

```suggestion
  description: A civilian grade weapon for hunting, or endless trench warfare. Uses .30 rifle ammo, equipped with standard-issue bayonet.
```

If you don't like that, you misspelt shiny.

```suggestion
  description: A civilian grade weapon for hunting, or endless trench warfare. Uses .30 rifle ammo. Now with new shiny bayonet!
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 2]
```diff
@@ -38,16 +38,60 @@
   - type: StaticPrice
     price: 500
 
-- type: entity
+- type: entity #Frontier
```
Comment (whatston3) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:41]: Suggesting commenting as shown below
```yaml
# Frontier - Mosin rework (PR#1391)
- type: entity
...
malus: 0.225
# End Frontier
```

**Review by whatston3 — CHANGES_REQUESTED** [review]
I like the changes, not entirely sure of the balance here, but being able to use the bayonet is very fun!  Looking forward to seeing this in-game.  Tests behaved as described, though I do think firing on RMB is odd.

If you could try testing this locally with [whatston3:2024-05-21_guns-melee-and-wielded](https://github.com/whatston3/frontier-station-14/tree/2024-05-21_guns-melee-and-wielded) merged in and setting useKey to false, I'd like to know your thoughts.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 3]
```diff
@@ -38,16 +38,60 @@
...
+  description: A civilian grade weapon for hunting, or endless trench warfare. Uses .30 rifle ammo. Now with new shiney bayonet!
   components:
   - type: Sprite
     sprite: Objects/Weapons/Guns/Snipers/bolt_gun_wood.rsi
-  - type: StaticPrice
-    price: 210
+    state: base
+  - type: AmmoCounter
+  - type: Gun
+    useKey: false
```
[the remark this answers is not in the document]
Comment (VividPups) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:52]: for the odd lmb animation, it became that when I moved some of the sound paths and Reorganized the placements of the melee component stuff. that was one of the effects of it  before it wasn't like that  but now it is

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml (hunk 2 above)
Comment (VividPups) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:41]: That actually sounds like a much better way to do it then from what I heard how it was done before.

**Comment by VividPups** [comment]
ALRIGHT,   since the merge that @Cheackraze  did to @whatston3 the Mosin is fully ready for testing.   The new ability is right-click is melee. left click is shooting, also added the ability to throw it,  this is without the ability to wield it

**Comment by VividPups** [comment]
https://youtu.be/37O_RzFNTZU

**Review by whatston3 — APPROVED** [review]
Seems fine locally, works as tested prior.  Not sure if others will use it, or of wider balance concerns, but glad to see the Mosin get some love, and it's a fun feature.

**Comment by dvir001** [comment]
@whatston3 Mosin is a bit broken right now working with one hand

**Comment by VividPups** [comment]
Can I ask how it's a bit broken?

**Comment by dvir001** [comment]
> @dvir001 Can I ask how it's a bit broken?

You can use the gun with one hand.

**Comment by VividPups** [comment]
> > @dvir001 Can I ask how it's a bit broken?
> 
> You can use the gun with one hand.

I wouldn't really say that's a bit broken, either way I can make it where it has to be wielded

**Comment by VividPups** [comment]
@dvir001   shooting and Melee now requires to be wielded in order to fire/melee attack.  also, Does these  "Conflicting files" have to do with me? Because I gone into the files it is saying it is in and its just the  nonlethal mosin  TDL stuff

**Comment by whatston3** [comment]
> dvir001 shooting and Melee now requires to be wielded in order to fire/melee attack. also, Does these "Conflicting files" have to do with me? Because I gone into the files it is saying it is in and its just the nonlethal mosin TDL stuff

The conflicts are here because the commented out non-lethal Mosin was migrated from Guns/Guns/snipers.yml to Guns/Snipers/snipers.yml, but your PR has a modification to Guns/Guns/snipers.yml.

Easiest way to fix this is to delete Guns/Guns/snipers.yml on your own repo.  The capacity should probably be removed from the non-lethal version in Guns/Snipers/snipers.yml, I think it'll be inherited from its parent.  Otherwise, just change it to 5 as well.

**Comment by VividPups** [comment]
all conflicts have been fix now

**Review by Cheackraze — APPROVED** [review]

**Pull request merged**